pinot-core/src/main/java/org/apache/pinot/core/util/fst/PinotBufferIndexInput.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.util.fst;
+
+import org.apache.lucene.store.IndexInput;
+import org.apache.pinot.core.segment.memory.PinotDataBuffer;
+
+import java.io.IOException;
+
+public class PinotBufferIndexInput extends IndexInput {
```

Comment:
Please add javadoc. Couple of other questions:

- Why do we need a specialized Pinotbuffer extending lucene buffer
- What is the arrangement of FST index in pinot segment directory? Is it under separate sub-directory like lucene text index. If the on-disk index is in a single buffer, can we combine it with other buffers in v3

Reply:
It's a wrapper so that lucene FST reader library could traverse over the buffer.
v1 generates a fst index file, v3 merges into the single index file now